File: javascript/026.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  if (nums.length === 0) return 0;

  let current = 0;
  let index = 0;
  let lastValue = nums[0] - 1;

  while(index < nums.length) {
    if (nums[index] !== lastValue) {
      nums[current] = nums[index];
      lastValue = nums[current];
      ++current;
      ++index;
    } else {
      ++index;
    }
  }

  return current;
};

File: ruby/026.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
  return 0 if nums.length == 0

  used_value = nums[0] - 1
  insert_index = 0
  0.upto(nums.length - 1) do |i|
    if nums[i] != used_value
      nums[insert_index] = nums[i]
      insert_index += 1
      used_value = nums[i]
    end
  end

  insert_index
end